influxql/engine.go

```diff
@@ -120,16 +111,26 @@ func (m *MapReduceJob) Execute(out chan *Row, filterEmptyResults bool) {
 				Name: m.MeasurementName,
 				Tags: m.TagSet.Tags,
 			}
-
 			return
 		} else {
-			pointCountInResult = m.stmt.Limit
-			if m.stmt.Offset+m.stmt.Limit > pointCountInResult {
-				pointCountInResult = pointCountInResult - m.stmt.Offset
+			if m.stmt.Offset+m.stmt.Limit > m.stmt.Limit {
+				if m.stmt.Limit-m.stmt.Offset < pointCountInResult {
```

Comment:
Yeah, this was not very clear to me.  I was looking for some comments or tests but could't find any to help determine some of the edge cases.  I'll write out some extended tests today to make sure we have all the edge cases covered, as well as add some documents in the code so future reviewers have a good grasp on what we are trying to accomplish.